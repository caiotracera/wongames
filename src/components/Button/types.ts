export type ButtonProps = {
  children?: React.ReactNode;
  size?: 'small' | 'medium' | 'large';
  fullWidth?: boolean;
  icon?: React.ReactNode;
  onClick?: () => (event: React.MouseEvent<HTMLButtonElement>) => void;
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

export type WrapperProps = {
  hasIcon?: boolean;
} & Pick<ButtonProps, 'size' | 'fullWidth'>;
