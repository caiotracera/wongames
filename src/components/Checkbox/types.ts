import { InputHTMLAttributes } from 'react';

export type CheckboxProps = {
  label?: string;
  labelFor?: string;
  labelColor?: 'white' | 'black';
  onCheck?: (status: boolean) => void;
  isChecked?: boolean;
  value?: string | ReadonlyArray<string> | number;
} & InputHTMLAttributes<HTMLInputElement>;

export type LabelProps = Pick<CheckboxProps, 'labelColor'>;
