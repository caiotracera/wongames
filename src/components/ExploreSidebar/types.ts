export type Field = {
  label: string;
  name: string;
};

export type ItemProps = {
  title: string;
  name: string;
  type: string;
  fields: Field[];
};

export type Values = {
  [field: string]: boolean | string;
};

export type ExploreSidebarProps = {
  items: ItemProps[];
  initialValues?: Values;
  onFilter: (values: Values) => void;
};

export type WrapperProps = {
  mobileFilterIsOpen?: boolean;
};
