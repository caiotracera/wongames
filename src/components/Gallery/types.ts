export type GalleryImageProps = {
  src: string;
  label: string;
};

export type GalleryProps = {
  items: GalleryImageProps[];
  color?: 'white' | 'black';
};

export type ModalProps = {
  isOpen: boolean;
};
