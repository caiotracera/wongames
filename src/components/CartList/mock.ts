import { GameItemProps } from '@/components/GameItem/types';

export const items: GameItemProps[] = [
  {
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    title: 'Red Dead Redemption 2',
    price: 'R$ 215,00',
    downloadLink: '/',
    paymentInfo: {
      flag: 'mastercard',
      img: '/img/cards/mastercard.png',
      number: '**** **** **** 1234',
      purchaseDate: '12/12/2020',
    },
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/150x70',
    title: 'Borderlands 3',
    price: 'R$ 215,00',
  },
];
