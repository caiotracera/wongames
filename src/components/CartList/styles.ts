import { tint } from 'polished';
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.white};
    display: flex;
    flex-direction: column;
    align-self: start;
  `}
`;

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${tint(0.2, theme.colors.lightGray)};

    color: ${theme.colors.black};
    font-weight: ${theme.font.weight.bold};
    font-size: ${theme.font.sizes.small};

    padding: 2rem;

    display: flex;
    justify-content: space-between;

    @media (min-width: 768px) {
      font-size: ${theme.font.sizes.medium};
      padding: ${theme.spacings.small};
    }
  `}
`;

export const Total = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
  `}
`;
