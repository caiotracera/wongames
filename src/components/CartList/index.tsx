import { GameItem } from '@/components/GameItem';

import { CartListProps } from '@/components/CartList/types';
import * as S from '@/components/CartList/styles';

export function CartList({ items, total }: CartListProps) {
  return (
    <S.Wrapper>
      {items.map((item) => (
        <GameItem key={item.title} {...item} />
      ))}

      <S.Footer>
        Total <S.Total>{total}</S.Total>
      </S.Footer>
    </S.Wrapper>
  );
}
