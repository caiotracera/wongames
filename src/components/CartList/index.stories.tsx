import { Story, Meta } from '@storybook/react/types-6-0';

import { CartList } from '@/components/CartList';
import { CartListProps } from '@/components/CartList/types';
import { items as cartListMock } from '@/components/CartList/mock';

export default {
  title: 'CartList',
  component: CartList,
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
  args: {
    items: cartListMock,
    total: 'R$ 430,00',
  },
} as Meta;

export const Default: Story<CartListProps> = (args) => (
  <div style={{ maxWidth: 800 }}>
    <CartList {...args} />
  </div>
);
