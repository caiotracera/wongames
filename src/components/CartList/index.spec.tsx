import { screen } from '@testing-library/react';

import { renderWithTheme } from '@/utils/tests/helpers';

import { CartList } from '@/components/CartList';
import { CartListProps } from '@/components/CartList/types';
import { items as cartListMock } from '@/components/CartList/mock';

const props: CartListProps = {
  items: cartListMock,
  total: 'R$ 430,00',
};

describe('<CartList />', () => {
  it('should render the <CartList />', () => {
    const { container } = renderWithTheme(<CartList {...props} />);

    expect(screen.getAllByRole('heading')).toHaveLength(2);
    expect(screen.getByText('R$ 430,00')).toHaveStyle({ color: '#F231A5' });

    expect(container.firstChild).toMatchSnapshot();
  });
});
