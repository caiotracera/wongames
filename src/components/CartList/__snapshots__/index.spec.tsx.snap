// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should render the <CartList /> 1`] = `
.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #eaeaea;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c3 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 1.4rem;
  font-weight: 600;
  color: #030517;
  line-height: 1.4rem;
  margin-bottom: 0.8rem;
}

.c8 {
  color: #fafafa;
  font-size: 1.4rem;
  font-weight: 600;
  padding: calc(0.8rem / 2) 0.8rem;
  border-radius: 0.4rem;
  background-color: #3CD3C1;
}

.c6 {
  color: #F231A5;
  margin-left: 0.8rem;
}

.c9 {
  color: #8f8f8f;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 28rem;
  margin-top: 1.6rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 img {
  margin-left: 0.8rem;
}

.c0 {
  background: #fafafa;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c11 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c12 {
  color: #F231A5;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c9 {
    margin-top: 0;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

@media (max-width:768px) {
  .c10 {
    margin-top: 1.6rem;
  }
}

@media (min-width:768px) {
  .c11 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="Red Dead Redemption 2"
          src="https://source.unsplash.com/user/willianjusten/151x70"
        />
      </div>
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Red Dead Redemption 2
          <a
            aria-label="Get Red Dead Redemption 2 here"
            class="c6"
            href="/"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="c7"
              fill="currentColor"
              focusable="false"
              height="22"
              viewBox="0 0 24 24"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 9h-4V3H9v6H5l7 8zM4 19h16v2H4z"
              />
            </svg>
          </a>
        </h3>
        <p
          class="c8"
        >
          R$ 215,00
        </p>
      </div>
    </div>
    <div
      class="c9"
    >
      <p>
        12/12/2020
      </p>
      <div
        class="c10"
      >
        <span>
          **** **** **** 1234
        </span>
        <img
          alt="mastercard"
          src="/img/cards/mastercard.png"
        />
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="Borderlands 3"
          src="https://source.unsplash.com/user/willianjusten/150x70"
        />
      </div>
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Borderlands 3
        </h3>
        <p
          class="c8"
        >
          R$ 215,00
        </p>
      </div>
    </div>
  </div>
  <div
    class="c11"
  >
    Total 
    <span
      class="c12"
    >
      R$ 430,00
    </span>
  </div>
</div>
`;
